starter-code/src/clue.js

```diff
@@ -1,94 +1,158 @@
+    color:        "red",
+    description:  "She is an A-list movie star with a dark past",
+    age:          31,
+    image:        "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+    occupation:   "Actor",
+    };
+    
+    var mrsPeacock = {
+    first_name:   "Eleanor",
+    last_name:    "Peacock",
+    color:        "blue",
+    description:  "She is from a wealthy family and uses her status and money to earn popularity",
+    age:          36,
+    image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+    occupation:   "Socialité",
+    };
+    
+    var mrMustard = {
+    first_name:   "Jack",
+    last_name:    "Mustard",
+    color:        "yellow",
+    description:  "He is a former football player who tries to get by on his former glory",
+    age:          62,
+    image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+    occupation:   "Retired Football player",
+    };
+    
+    var Weapon1 = {
+    name: "rope", weight: 10,
+    };
+    var   Weapon2 = {
+    name: "knife",weight: 8,};
+    var   Weapon3 = {
+    name: "candlestick",weight: 2,
+    };
+    var   Weapon4 = {
+    name: "dumbbell" ,weight: 30,
+    };
+    var   Weapon5 = {
+    name: "poison", weight: 2,
+    };
+    var   Weapon6= {
+    name: "axe", weight: 15,
+    };
+    var   Weapon7 = {
+    name: "bat",weight: 13,
+    };
+    var   Weapon8 = {
+    name: "trophy" , weight: 25,
+    };
+    var   Weapon9= {
+    name: "pistol",weight: 20,
+    };
+    
+    // // Rooms
+    var room1 = {
+    name: "Dinning Room"};
+    var room2 = {
+    name: "Conservatory"};
+    var room3 = {
+    name: "Kitchen"};
+    var room4 = {
+    name: "Study"};
+    var room5 = {
+    name: "Library"};
+    var room6 = {
+    name: "Billiard Room"};
+    var room7 = {
+    name: "Lounge"};
+    var room8 = {
+    name: "Ballroom"};
+    var room9 = {
+    name: "Hall"};
+    var room10 = {
+    name: "Spa"};
+    var room11 = {
+    name: "Living Room"};
+    var room12 = {
+    name: "Observatory"};
+    var room13 = {
+    name: "Theater"};
+    var room14 = {
+    name: "Guest House"};
+    var room15 = {
+    name: "Patio"};
+    
+    
+    // Characters Collection
+    var charactersArray = [ mrGreen ,drOrchid,profPlum ,missScarlet,mrsPeacock,mrMustard ];
+    //console.log(charactersArray);
+    
+    // Rooms' Collection
+    var roomsArray = [room1,room2,room3,room4,room5,room6,room7,room8,room9,room10,room11,room12,room13,room14,room15];
+    //console.log(roomsArray);
+    
+    // Weapons Collection
+    var weaponsArray = [
+    Weapon1,Weapon2,Weapon3,Weapon4,Weapon5,Weapon6,Weapon7,Weapon8,Weapon9];
+    //console.log(weaponsArray);
+    
+    function randomSelector(a){
+    // for(i=0;i<a.length;i++){
+    var i = Math.floor(Math.random()* (a.length-1));
+    return a[i];
+    }
+    //console.log(randomSelector(weaponsArray));
+    
+    function pickMistery (a,b,c){
```

Comment:
Jasmine test is very picky so we have to give it what it asks :)
there are not arrays passed as parameters.
This would be the solution:
```
function pickMistery (){
      var misteryEnvelope = [];
      misteryEnvelope.push(randomSelector(charactersArray));
      misteryEnvelope.push(randomSelector(weaponsArray));
      misteryEnvelope.push(randomSelector(roomsArray));
      return misteryEnvelope;
    }